src/main/scala/com/thoughtworks/enableIf.scala

```diff
@@ -3,8 +3,54 @@ package com.thoughtworks
 import scala.annotation.StaticAnnotation
 import scala.reflect.internal.annotations.compileTimeOnly
 import scala.reflect.macros.Context
+import scala.util.matching.Regex
+
 
 object enableIf {
+  private def getRegex(artifactId: String, regex: Regex): String = {
+    s".*${artifactId}-${regex.toString}"
+  }
+
+  private def getRegex(artifactId: String, version: String): String = {
+    val versionRegex = s"${version.replace(".", "\\.")}.*"
+    getRegex(artifactId, new Regex(versionRegex))
+  }
+
+  private def getScalaRegex(artifactId: String, regex: Regex): String = {
+    val scalaMajorVersion = scala.util.Properties.versionNumberString.split("\\.")
+      .take(2).mkString(".")
+    getRegex(s"${artifactId}_${scalaMajorVersion}", regex)
+  }
+
+  private def getScalaRegex(artifactId: String, version: String): String = {
+    val versionRegex = s"${version.replace(".", "\\.")}.*"
+    getScalaRegex(artifactId, new Regex(versionRegex))
+  }
+
+  def classpathMatches(regex: String): Context => Boolean = {
+    c => c.classPath.exists(_.getPath.matches(regex))
+  }
+
+  def classpathMatches(regex: Regex): Context => Boolean = {
+    c => c.classPath.exists(_.getPath.matches(regex.toString))
```

Comment:
Don't call `regex.toString`. It makes no sense to compile a regex then decompile it then compile it again.

Reply:
regex.toString does not compile, it only returns the internal regex string.

Reply:
Write a regex like `".*xyz.*".r`, IDE will help us to write the correct regex string. That's why I'm using Regex as the param here.

Reply:
<img width="455" alt="image" src="https://user-images.githubusercontent.com/66561265/160051825-bee2e1f2-b808-400f-9aae-878f648a0ff8.png">

Reply:
`.r` is compilation

Reply:
Fixed.